homeassistant/components/jewish_calendar/sensor.py

```diff
@@ -1,140 +1,60 @@
-    "upcoming_shabbat_candle_lighting": [
-        "Upcoming Shabbat Candle Lighting",
-        "mdi:candle",
-    ],
-    "upcoming_shabbat_havdalah": ["Upcoming Shabbat Havdalah", "mdi:weather-night"],
-    "upcoming_candle_lighting": ["Upcoming Candle Lighting", "mdi:candle"],
-    "upcoming_havdalah": ["Upcoming Havdalah", "mdi:weather-night"],
-    "issur_melacha_in_effect": ["Issur Melacha in Effect", "mdi:power-plug-off"],
-    "omer_count": ["Day of the Omer", "mdi:counter"],
-}
-
-CONF_DIASPORA = "diaspora"
-CONF_LANGUAGE = "language"
-CONF_SENSORS = "sensors"
-CONF_CANDLE_LIGHT_MINUTES = "candle_lighting_minutes_before_sunset"
-CONF_HAVDALAH_OFFSET_MINUTES = "havdalah_minutes_after_sunset"
-
-CANDLE_LIGHT_DEFAULT = 18
-
-DEFAULT_NAME = "Jewish Calendar"
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_DIASPORA, default=False): cv.boolean,
-        vol.Optional(CONF_LATITUDE): cv.latitude,
-        vol.Optional(CONF_LONGITUDE): cv.longitude,
-        vol.Optional(CONF_LANGUAGE, default="english"): vol.In(["hebrew", "english"]),
-        vol.Optional(CONF_CANDLE_LIGHT_MINUTES, default=CANDLE_LIGHT_DEFAULT): int,
-        # Default of 0 means use 8.5 degrees / 'three_stars' time.
-        vol.Optional(CONF_HAVDALAH_OFFSET_MINUTES, default=0): int,
-        vol.Optional(CONF_SENSORS, default=["date"]): vol.All(
-            cv.ensure_list, vol.Length(min=1), [vol.In(SENSOR_TYPES)]
-        ),
-    }
-)
+_LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Jewish calendar sensor platform."""
-    language = config.get(CONF_LANGUAGE)
-    name = config.get(CONF_NAME)
-    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
-    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
-    diaspora = config.get(CONF_DIASPORA)
-    candle_lighting_offset = config.get(CONF_CANDLE_LIGHT_MINUTES)
-    havdalah_offset = config.get(CONF_HAVDALAH_OFFSET_MINUTES)
-
-    if None in (latitude, longitude):
-        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
+    if discovery_info is None:
         return
 
-    dev = []
-    for sensor_type in config[CONF_SENSORS]:
-        dev.append(
-            JewishCalSensor(
-                name,
-                language,
-                sensor_type,
-                latitude,
-                longitude,
-                hass.config.time_zone,
-                diaspora,
-                candle_lighting_offset,
-                havdalah_offset,
-            )
-        )
-    async_add_entities(dev, True)
+    async_add_entities(
+        [
+            JewishCalendarSensor(hass.data[DOMAIN], sensor, sensor_info)
+            for sensor, sensor_info in SENSOR_TYPES["data"].items()
+        ]
+    )
 
+    async_add_entities(
```

Comment:
Please collect all entities first and make a single call to `async_add_entities`.